lib/Net/SAML2/Binding/SOAP.pm

```diff
@@ -152,13 +172,15 @@ sub handle_response {
     my ($self, $response) = @_;
 
     my $saml = _get_saml_from_soap($response);
-    $self->verify_xml(
-        $saml,
-        no_xml_declaration => 1,
-        cert_text          => $self->idp_cert,
-        cacert             => $self->cacert,
-        anchors            => $self->anchors
-    );
+    foreach my $cert (@{$self->idp_cert}) {
```

Comment:
This should be in a try/catch as `verify_xml` croaks on error. If `verify_xml` doesn't croak the `request` function should return the XML (eg, add a last LOOP). It should check the response on all the remaining certs. When none is able to verify the XML the function should croak.

Reply:
Still not happy here - see the note on 5.14 eval may be better for now - also not sure what to return as error if all certs fail.  Do we care at all of one failed as long as one succeeded?